### PR TITLE
Fix for issue #125 Multiple users are able to sign up with the same email

### LINKED ISSUE #125: Multiple users are able to sign up with the same email
The email that users can use for sign-up does not have a `unique` check and multiple users are able to sign in using the same Email Id. Since this is a security risk.

### PR DESCRIPTION
**Pull request by theonly1me**
fix: prevent multiple users from registering via the same email id

## Description

Previously multiple users could use the same email address to sign up. With this fix, an email can only be associated to a single user

## Related Issue
Fix #125

### CODE REVIEW

File: src/client/routes/signup/index.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import Success from '../../components/info/success';
 import { signUp } from '../../api/authentication';
 import { setToken } from '../../helpers/token';
 import config from '../../config';
+import { userLogin } from '../../actions';
```
Comment (theonly1me) [inline src/client/routes/signup/index.js:12]: This import somehow did not go through in my last PR, sorry about that. That is why I have added it here.

File: src/server/services/redis.js [hunk 2]
```diff
@@ -128,6 +128,16 @@ export async function getUser(username) {
     return client.hgetall(`user:${username}`);
 }
 
+export async function getAllUsers() {
+    const usernames = await getAllUserNames();
+    return Promise.all(usernames.map(async (username) => await client.hgetall(`user:${username}`)));
+}
+
+export async function getAllUserNames() {
+    const keys = await client.keys('*');
```
Comment (bjarneo) [inline src/server/services/redis.js:137]: Try to use `client.keys('user:*');` instead, I think that will work

**Review by bjarneo — CHANGES_REQUESTED** [review]
A minor suggestion. Else it looks great to me.

I think we should get that sqlite database up and running for the user interaction sooner than later, because to find a user by its value in a hset, that is not straight forward 😅 If anything it should be sorted sets, but, this is me just rambling about what should have been. Someone is looking into the sqlite database, so that is great

File: src/server/services/redis.js (hunk 2 above)
Comment (theonly1me) [reply src/server/services/redis.js]: I have changed it to be `client.keys('user:*');` :)

**Comment by theonly1me** [comment]
> I think we should get that sqlite database up and running for the user interaction sooner than later, because to find a user by its value in a hset, that is not straight forward 😅 

Totally agree with this since the key for hset is currently the user name, having any other db that will allow us to query on basis of document (no-sql) or column (sql) data would make it easier overall. Also, Redis might not be enough at scale

**Comment by bjarneo** [comment]
> > I think we should get that sqlite database up and running for the user interaction sooner than later, because to find a user by its value in a hset, that is not straight forward 😅
> 
> Totally agree with this since the key for hset is currently the user name, having any other db that will allow us to query on basis of document (no-sql) or column (sql) data would make it easier overall. Also, Redis might not be enough at scale

Oh no, it wont scale. Started out just for secrets, but you know. It grows. 


Thanks for this PR :)

**Review by bjarneo — APPROVED** [review]

**Pull request merged**